torch/csrc/jit/passes/peephole.cpp

```diff
@@ -46,9 +46,11 @@ void PeepholeOptimizeImpl(Block* block, bool addmm_fusion_enabled) {
       // x.expand(x.size()) == x
       if (auto input_type = node->namedInput(attr::self)
                                 ->type()
-                                ->cast<CompleteTensorType>()) {
+                                ->cast<ProfiledTensorType>()) {
         auto expanded_sizes = node->get<c10::List<int64_t>>(attr::size);
-        if (!expanded_sizes.has_value() || c10::impl::toVector(*expanded_sizes) == input_type->sizes()) {
+        auto input_type_sizes = input_type->sizes().concrete_sizes();
+        if (expanded_sizes.has_value() && input_type_sizes &&
```

Comment:
is omitting `!` in front of `expanded_sizes.has_value()` intentional?

Reply:
yes, my reading was that this code was completely bogus before this change. Before it said 'if expanded_sizes is not a constant, then it is safe to remove the expand call'. Now it says 'if expanded_sizes is a constant, and  its value is the same as the input sizes, we can remove the expand'.